analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedPrivateMember.cs

```diff
@@ -420,3 +420,17 @@ class Repro_8348
 {
     [MyAttribute] void PrivateMethodWithAttribute() { } // FN: due to the attribute
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8342
+class Repro_8342
+{
+    [Private1] public void APublicMethod() => APrivateMethodCalledByAPublicMethod();
```

Comment:
Maybe it's also worth adding a test case with multiple attributes (and list attributes):
```cs
[SomePublicAttribute, Private1]
public void Something() { }

[SomePublicAttribute]
[Private1]
public void Something2() { }
```